blas/unit_test/Test_Blas2_syr.hpp

```diff
@@ -0,0 +1,1725 @@
+                              !std::is_same<T, Kokkos::complex<double>>::value,
+                          void>::type
+  compareVanillaAgainstExpected(const T& alpha,
+                                const _ViewTypeExpected& h_vanilla,
+                                const _ViewTypeExpected& h_expected);
+
+  template <class T>
+  typename std::enable_if<std::is_same<T, Kokkos::complex<float>>::value ||
+                              std::is_same<T, Kokkos::complex<double>>::value,
+                          void>::type
+  compareKkSyrAgainstReference(const T& alpha, const _HostViewTypeA& h_A,
+                               const _ViewTypeExpected& h_reference);
+
+  template <class T>
+  typename std::enable_if<!std::is_same<T, Kokkos::complex<float>>::value &&
+                              !std::is_same<T, Kokkos::complex<double>>::value,
+                          void>::type
+  compareKkSyrAgainstReference(const T& alpha, const _HostViewTypeA& h_A,
+                               const _ViewTypeExpected& h_reference);
+
+  template <class T>
+  T shrinkAngleToZeroTwoPiRange(const T input);
+
+  template <class TX>
+  void callKkSyrAndCompareAgainstExpected(const ScalarA& alpha, TX& x,
+                                          _ViewTypeA& A,
+                                          const _HostViewTypeA& h_A,
+                                          const _ViewTypeExpected& h_expected,
+                                          const std::string& situation);
+
+  template <class TX>
+  void callKkGerAndCompareKkSyrAgainstIt(
+      const ScalarA& alpha, TX& x,
+      view_stride_adapter<_ViewTypeA, false>& org_A,
+      const _ViewTypeExpected& h_A_syr, const std::string& situation);
+
+  const bool _A_is_complex;
+  const bool _A_is_lr;
+  const bool _A_is_ll;
+  const bool _testIsGpu;
+  const bool _vanillaUsesDifferentOrderOfOps;
+  const _AuxType _epsAbs;
+  const _AuxType _epsRel;
+  int _M;
+  int _N;
+  bool _useAnalyticalResults;
+  bool _useHermitianOption;
+  bool _useUpOption;
+  bool _kkSyrShouldThrowException;
+  bool _kkGerShouldThrowException;
+};
+
+template <class ScalarX, class tLayoutX, class ScalarA, class tLayoutA,
+          class Device>
+SyrTester<ScalarX, tLayoutX, ScalarA, tLayoutA, Device>::SyrTester()
+    : _A_is_complex(std::is_same<ScalarA, Kokkos::complex<float>>::value ||
+                    std::is_same<ScalarA, Kokkos::complex<double>>::value),
+      _A_is_lr(std::is_same<tLayoutA, Kokkos::LayoutRight>::value),
+      _A_is_ll(std::is_same<tLayoutA, Kokkos::LayoutLeft>::value),
+      _testIsGpu(KokkosKernels::Impl::kk_is_gpu_exec_space<
+                 typename Device::execution_space>())
+#ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
+      ,
+      _vanillaUsesDifferentOrderOfOps(_A_is_lr)
+#else
+      ,
+      _vanillaUsesDifferentOrderOfOps(false)
+#endif
+      ,
+      _epsAbs(std::is_same<_AuxType, float>::value ? 1.0e-6 : 1.0e-9),
```

Comment:
Please provide some explanation for these magic numbers for future readers, even if there's not a good explanation :)

Reply:
Done.